goweb/src/auth/SupabaseAuthProvider.tsx

```
@@ -1,157 +1,254 @@
-          if (session?.user && !hasUserBeenSynced(session.user.id)) {
-            await ensureUserInBackend(session.user, session.access_token);
-          }
-        }
+        // Check if we have session in sessionStorage (for hot reloads)
+        const cachedSessionStr = sessionStorage.getItem(
+          "supabase_auth_session"
+        );
+        if (cachedSessionStr) {
+          try {
+            const cachedSession = JSON.parse(cachedSessionStr);
+            if (cachedSession && cachedSession.user) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)